### PR TITLE
feat(typescript): add ICollectionObserverSplice definition

### LINKED ISSUE #652: Missing TypeScript Interface: ICollectionObserverSplice
I have to add this interface to all of my projects. It would be nice if the framework exported it for us. 

```
interface ICollectionObserverSplice {
    addedCount: number;
    index: number;
    removed: Array<any>;
}
```

This would be used in `BindingEngine.collectionObserver(splices: ICollectionObserverSplice)`

### PR DESCRIPTION
**Pull request by jonathaneckman**
Closes #652

### CODE REVIEW

**Comment by jonathaneckman** [comment]
May also complete part of #607.

**Comment by EisenbergEffect** [comment]
Nice work @jonathaneckman ! Also, thanks for writing a blog post on your experience. I'm actually planning to create some more contributor docs soon, placed on the main site so they are easier to find. I'd love to have you contribute to those docs to help craft something that will help new contributors. Perhaps your blog post can serve as a starting point?

@jdanyow Can you review the PR above and ensure that the typings are correct. They look good to me :)

**Comment by jonathaneckman** [comment]
@EisenbergEffect I'd be happy to help! Point me in the right direction and I'll get it started.

**Comment by EisenbergEffect** [comment]
@jonathaneckman We have all our docs on GitHub. Each library has a "docs" folder and under that there will be articles on the library. Not every repo has that. Most of it is in the "framework" repo. If you feel inclined and want to write some documentation on contributing to the project, you could just PR it to the docs/articles folder in the framework repo and you and I could review it together and then merge it in. Naturally, your name would appear on the article so you would get credit for the work. We can coordinate it with an official blog on our site as well to get you some exposure (if you want) and share with the community about our improved contributions resources.

Basically, we'd love to have you help out and I'll do what I can to make it easy and worth it for you.

**Comment by jonathaneckman** [comment]
@EisenbergEffect Sounds great. I'll pull down the framework docs and get my bearings this weekend. Continuing on what I started with this PR, I could write a guide explaining how to observe collections so we can close #607. If the content is there, I may also follow up with blog post tailored to documentation contributions.

**Comment by EisenbergEffect** [comment]
Sounds great!

**Comment by jonathaneckman** [comment]
@EisenbergEffect I started on these docs, but have questions. Is there a way to test or view the rendered page myself prior to submission? I assume you use something like docfx. You can see the work in progress [here](https://github.com/jonathaneckman/binding/blob/features/observable-docs/doc/article/en-US/binding-observables.md), but its not even first draft yet. 

@jdanyow I may have another PR coming your way soon. I didn't realize that the splice interface may change depending on the type of collection you subscribed to. To learn, I've started throwing together a [demo project](https://github.com/jonathaneckman/demos/tree/master/aurelia-collection-observer/src/routes) to test the various collections.

**Comment by EisenbergEffect** [comment]
There isn't an easy way to preview what it will look like on the site. If you use basic Markdown as you are doing, it should be fine. I test it myself and am happy to make any fixes that are needed when it's ready.

There are a couple of extensions to Markdown. You can do an Info, Warning or Danger text on the first line of a block quote and then you'll get some special styles. You can also do "Info: Some Title" if you want a special title.

It's nice to have esNext and TypeScript if they are different but not required. We can add that later.

No big changes planned other than maybe adding some new things. I tried to make a lot of the big changes over the last month.
